src/machine/machinectl.c

```diff
@@ -1412,6 +1414,149 @@ static int shell_machine(int argc, char *argv[], void *userdata) {
+                                                       *machine);
+
+                        if (is_var) {
+                                _cleanup_free_ char *file = NULL, *new_path = NULL;
+
+                                file = strjoin(*machine, ".nspawn");
+                                if (!file)
+                                        return log_oom();
+
+                                new_path = path_join("/etc/systemd/nspawn", file);
+                                if (!new_path)
+                                        return log_oom();
+
+                                r = edit_files_add(&context, new_path, path, NULL);
+                        } else
+                                r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                }
+
+        } else {
+                STRV_FOREACH(machine, argv + 1) {
+                        _cleanup_free_ char *path = NULL, *file = NULL;
+
+                        r = get_settings_path(*machine, &path, NULL);
+                        if (r >= 0)
+                                return log_error_errno(SYNTHETIC_ERRNO(EEXIST),
+                                                       "Settings for the specified machine '%s' exists in '%s', refusing.",
+                                                       *machine,
+                                                       path);
+
+                        file = strjoin(*machine, ".nspawn");
+                        if (!file)
+                                return log_oom();
+
+                        path = path_join("/etc/systemd/nspawn", file);
+                        if (!path)
+                                return log_oom();
+
+                        r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                }
+        }
+
+        return do_edit_files_and_install(&context);
+}
+
+static int cat_settings(int argc, char *argv[], void *userdata) {
+        int r = 0;
+
+        if (!IN_SET(arg_transport, BUS_TRANSPORT_LOCAL, BUS_TRANSPORT_MACHINE))
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                       "Cat is only supported on local machines.");
+
+        pager_open(arg_pager_flags);
+
+        STRV_FOREACH(machine, argv + 1) {
+                _cleanup_free_ char *path = NULL;
+                int q;
+
+                q = get_settings_path(*machine, &path, NULL);
+                if (q == -ENOENT) {
+                        log_error_errno(q, "No settings file found for machine '%s'.", *machine);
+                        r = r < 0 ? r : q;
+                        continue;
+                }
+                if (q < 0)
+                        return r < 0 ? r : q;
+
+                q = cat_files(path, 0, 0);
+                if (q < 0)
+                        return r < 0 ? r : q;
```

Comment:
Also here.

Reply:
This and the one above have been logged by the callee I believe 🤔

Reply:
Indeed. `cat_files()` logs its failure.